web/src/containers/admin/CreateAccount.js

```diff
@@ -1,24 +1,46 @@
+import { FormLabel, Select, TextField } from '@cmsgov/design-system-core';
 import PropTypes from 'prop-types';
 import React, { Component, Fragment } from 'react';
 import { connect } from 'react-redux';
-import { Link } from 'react-router-dom';
-import { STATES } from '../../util';
+import CardForm from '../../components/CardForm';
+import Header from '../../components/Header';
 import Password from '../../components/PasswordWithMeter';
-
+import { STATES } from '../../util';
 import { createUser as createUserDispatch } from '../../actions/admin';
-
-import Btn from '../../components/Btn';
-import { t } from '../../i18n';
+import { getAddAccountError } from '../../reducers/errors';
+import { getAddAccountWorking } from '../../reducers/working';
 
 class CreateUser extends Component {
   state = {
-    fetching: false,
+    hasFetched: false,
     name: '',
     email: '',
     password: '',
-    state: ''
+    role: '',
+    state: '',
+    success: false
   };
 
+  static getDerivedStateFromProps({ error, working }, { hasFetched }) {
+    if (!hasFetched) {
+      return { hasFetched: working };
+    }
+
+    if (!working && !error) {
+      return {
+        hasFetched: false,
+        name: '',
+        email: '',
+        password: '',
+        role: '',
+        state: '',
+        success: true
+      };
+    }
+
+    return { success: false };
+  }
```

Comment:
We have to derive whether or not the form was successfully submitted.  It can't be stored in the app state because then the next time this form was loaded, it would default to a success state, which feels wrong.  It's kinda hacky logic, but it doesn't feel too awful.  Like, it doesn't make me do this:

![](https://media.giphy.com/media/aCpItaoS1gjg4/giphy.gif)

Reply:
Could we throw a comment in here to this tune? This seems like the kind of thing that someone will stumble on six months from now and be confused about. (Hi, Future Heather.)

Reply:
Done!